src/inmanta/data/__init__.py

```diff
@@ -1218,26 +1228,30 @@ async def _expire_endpoints_for_process(
 
     @classmethod
     async def log_instance_creation(
-        cls, tid: uuid.UUID, process: uuid.UUID, endpoints: Set[str], now: datetime.datetime
+        cls: Type[TAgentInstance], tid: uuid.UUID, process: uuid.UUID, endpoints: Set[str], now: datetime.datetime
     ) -> None:
         """
         Create new agent instances for a given session.
         """
-        if not endpoints:
-            return
-        # Fix database corruption when database was down
-        await cls._expire_endpoints_for_process(tid, process, endpoints, now)
```

Comment:
@arnaudsjs will have to check if this can be removed

Reply:
tests seem to think this is still required ;-)

Reply:
I noticed. My reasoning was that we would now reuse documents instead of expiring them and creating a new one. But I might be missing something, I'll look into it.

Reply:
Indeed. That's still required.

Reply:
Actually, I think what I did above is not sufficient. When an agent disconnects the corresponding `AgentInstance` document will still be set to expired. A call to `log_instance_creation` will then fail to find an active instance and will create a new one. This, combined with @wouterdb's comment (on slack) to enforce this in the DB, makes me think the best way to solve this would be to allow only documents with a distinct `(tid, process, name)` tuple. `log_instance_creation` would then just unexpire an expired instance document if it exists. It looks like the id for the agent instance document is not used in code or as a foreign key, so I think this behavioral change shouldn't impact anything.
@arnaudsjs do you agree this would be the best approach? Do you also agree that in this case the call to `_expire_endpoints_for_process` can be safely removed?

Reply:
(an index on `(tid, process, name)`, I couldn't think of the word, which is surprising in hindsight because it's the same in the inmanta DSL.

Reply:
Yes. I think you are right, `_expire_endpoints_for_process` can be safely removed in that case.